src/rest/connector/tests/test_nxos.py

```diff
@@ -362,3 +358,19 @@ def test_request_exception_retry(self):
             connection.disconnect()
 
         self.assertEqual(connection.connected, False)
+
+    def test_headers(self):
+        connection = Rest(device=self.device, alias='rest', via='rest')
+        self.assertEqual(connection.connected, False)
+
+        with patch('requests.Session') as req:
+            resp = Response()
+            resp.status_code = 200
+            req().post.return_value = resp
+            connection.connect()
+            resp.json = MagicMock(return_value={'imdata': []})
+            connection.post(
+                dn='temp', payload={'payload': 'something'}, headers='str'
+            )
+            connection.disconnect()
+        self.assertEqual(connection.connected, False)
```

Comment:
This test does not make sense after disconnect().  You are passing headers as a string, are you expecting to raise a ValueError for this?

Reply:
I couldn't raise an exception. I'm just using default values once headers is not a dict like the other implementations.

Reply:
Please raise a ValueError in case the `headers` object passed by the user is not a dictionary